composer/trainer/trainer.py

```diff
@@ -3461,6 +3462,8 @@ def _eval_loop(
                     try:
                         microbatches = data_spec.split_batch(device_batch, evaluator.device_eval_microbatch_size)
                         for i, self.state.batch in enumerate(microbatches):
+                            self.state.batch = self.state.device.batch_to_device(self.state.batch)
+                            self.state.batch = data_spec.device_transforms(self.state.batch)
```

Comment:
isnt this already called in the outer for loop and therefore is not needed here? it should probably only be called once right

Reply:
aren't some non-default, custom `device_transforms` methods device dependent? So doing it on GPU vs CPU would matter? I think it makes sense to get rid of the first `device_transforms` outside of the for loop

Reply:
Agree, lets do that